client/state/reader/posts/reducer.js

```diff
@@ -20,7 +20,19 @@ export function items( state = {}, action ) {
 	switch ( action.type ) {
 		case READER_POSTS_RECEIVE:
 			const posts = action.posts || action.payload.posts;
-			return { ...state, ...keyBy( posts, 'global_ID' ) };
+			const postsByDedupedGlobalId = {};
+
+			// Assign a new key in case of global_ID collision.
+			for ( const [ globalId, post ] of Object.entries( keyBy( posts, 'global_ID' ) ) ) {
+				let key = globalId;
+				if ( state[ globalId ] && state[ globalId ].feed_item_ID !== post.feed_item_ID ) {
+					key = `${ globalId }-${ post.feed_item_ID }`;
+				}
+
+				postsByDedupedGlobalId[ key ] = post;
```

Comment:
Does it affect the `seen` feature below as it seems to use the `global_ID` to mark whether the post is seen or not 🤔

Reply:
Good point... let me think of another approach.